File: mobile_app/lib/models/coffee_bean.dart
class CoffeeBean {
  final int id;
  final String name;
  final String origin;
  final String variety;
  final String roastLevel;
  final String? tastingNotes;

  CoffeeBean({
    required this.id,
    required this.name,
    required this.origin,
    required this.variety,
    required this.roastLevel,
    this.tastingNotes,
  });

  factory CoffeeBean.fromJson(Map<String, dynamic> json) {
    return CoffeeBean(
      id: json['id'] as int,
      name: json['name'] as String,
      origin: json['origin'] as String,
      variety: json['variety'] as String,
      roastLevel: json['roast_level'] as String,
      tastingNotes: json['tasting_notes'] as String?,
    );
  }
}

File: packs/beans/app/models/coffee_bean.rb
module Beans
  class CoffeeBean
    attr_accessor :name, :origin, :variety, :roast_level, :tasting_notes, :purchased_at, :store_id, :user_id

    def initialize(name:, origin:, variety:, roast_level:, tasting_notes: nil, purchased_at: nil, store_id: nil, user_id: nil)
      @name = name
      @origin = origin
      @variety = variety
      @roast_level = roast_level
      @tasting_notes = tasting_notes
      @purchased_at = purchased_at
      @store_id = store_id
      @user_id = user_id
    end
  end
end
